aec_checker.py

```diff
@@ -61,42 +117,52 @@ def getAECStatus(
 
     elem = driver.find_element(By.ID, "ctl00_ContentPlaceHolderBody_textStreetName")
     elem.clear()
-    elem.send_keys(streetName)
+    elem.send_keys(street)
 
-    captcha_entered = False
+    captcha_passed = False
+    success_panel = None
+    send_driver_to_captcha(driver)
 
-    driver.find_element(
-        By.ID, "ctl00_ContentPlaceHolderBody_textVerificationCode"
-    ).send_keys("")
-
-    while not captcha_entered:
-        time.sleep(1)
+    while not success_panel and not captcha_passed:
+        time.sleep(0.1)
         try:
-            elem = driver.find_element(
-                By.ID, "ctl00_ContentPlaceHolderBody_textVerificationCode"
+            captcha_elem = driver.find_element(
+                By.ID, CAPTCHA_INPUT_ID
             )
         except NoSuchElementException:
+            print("The browser must have been commandeered")
             break
 
-        if len(elem.get_attribute("value")) == 4:
+        captcha_value = captcha_elem.get_attribute("value")
+        if len(captcha_value) == 4:
             driver.find_element(
                 By.ID, "ctl00_ContentPlaceHolderBody_buttonVerify"
             ).click()
+            print("The button has been clicked to verify the CAPTCHA")
+            time.sleep(0.1)  # Wait for submission to be executed
 
-            try:
-                # Look for the first name tag, if it exist the captcha failed
```

Comment:
The CAPTCHA might not have failed; it could be that the new page hadn't finished loading